model/ftn/wmesmfmd.ftn

```diff
@@ -656,13 +692,62 @@
         ': entered InitializeP1', ESMF_LOGMSG_INFO)
 !
 ! -------------------------------------------------------------------- /
+! Query mediator specific attributes 
+!
+      if (med_present) then
+        call NUOPC_CompAttributeGet(gcomp, name="ScalarFieldName", &
+          value=cvalue, isPresent=isPresent, isSet=isSet, rc=rc)
+        if (ESMF_LogFoundError(rc, PASSTHRU)) return
+        if (isPresent .and. isSet) then
+          flds_scalar_name = trim(cvalue)
+          call ESMF_LogWrite(trim(cname)//': flds_scalar_name = '// &
+            trim(flds_scalar_name), ESMF_LOGMSG_INFO, rc=rc)
+          if (ESMF_LogFoundError(rc, PASSTHRU)) return
+        end if
+
+        call NUOPC_CompAttributeGet(gcomp, name="ScalarFieldCount", &
+          value=cvalue, isPresent=isPresent, isSet=isSet, rc=rc)
+        if (ESMF_LogFoundError(rc, PASSTHRU)) return
+        if (isPresent .and. isSet) then
+          read(cvalue, *) flds_scalar_num
```

Comment:
My preference for converting strings is to use the ESMF conversion utility (i.e. ESMF_UtilString2Int)
https://earthsystemmodeling.org/docs/release/ESMF_8_1_1/ESMF_refdoc/node6.html#SECTION060144800000000000000